dsew_community_profile/delphi_dsew_community_profile/pull.py

```diff
@@ -628,8 +684,8 @@ def add_max_ts_col(df):
     """
     Add column to differentiate timestamps for a given publish date-geo combo.
 
-    Each publish date is associated with two timestamps for test volume and
-    test positivity. The older timestamp corresponds to data from the
+    Each publish date is associated with up to two timestamps for test volume
+    and test positivity. The older timestamp corresponds to data from the
```

Comment:
does this change mean that you found files that only have one week of test data in them, instead of both "last week" and "previous week"?

Reply:
In short, no, this is the bit that actually fixes issue #1556.

Previously, we expected that "[t]esting signals should have two unique timestamps per publish date-region combination." This is true of _raw_ data, but causes false positives for our processed data.

This particular check [is called in `unify_testing_sigs`](https://github.com/cmu-delphi/covidcast-indicators/blob/eaeda1d87b644726cc2f997cbb80db3869653732/dsew_community_profile/delphi_dsew_community_profile/pull.py#L558-L559) after we [deduplicate timestamps](https://github.com/cmu-delphi/covidcast-indicators/blob/eaeda1d87b644726cc2f997cbb80db3869653732/dsew_community_profile/delphi_dsew_community_profile/pull.py#L518-L525). We only keep those with the most recent publish date, which means that some publish date-timestamp combinations are dropped.

Specifically, if we're generating output over a wide enough time period so that we see a given date reported for "last week" and also for the "previous week", we will always keep the version reported for the "previous week" since it's publish date will be 7 days newer. As a result, the older publish date will have all of it's "last week" data discarded; it will only be associated with the "previous week"'s timestamp.

Reply:
Are there ever cases where only one timestamp is available for volume but two are available for positivity, or vice-versa?

Reply:
* checks that each publish date has the same geos for test volume and test positivity [here](https://github.com/cmu-delphi/covidcast-indicators/blob/ndefries/cpr-lenient-check-ts-per-publishdate/dsew_community_profile/delphi_dsew_community_profile/pull.py#L646-L648)
* checks that test volume and positivity have the same number of observations. Since we've already checked that they have the same geos by publish date, this ~checks that they have the same number of ts per publish date-geo.

Reply:
Assuming there are always the same number of timestamps available for test volume and positivity in the raw data, there should always be the same number in the processed data. The only logic that influences this is the timestamp deduplication (where we keep the most recent publish date for each timestamp). It is [applied to both test volume and positivity](https://github.com/cmu-delphi/covidcast-indicators/blob/ndefries/cpr-lenient-check-ts-per-publishdate/dsew_community_profile/delphi_dsew_community_profile/pull.py#L518-L523) using the same ["change date" threshold approach](https://github.com/cmu-delphi/covidcast-indicators/blob/4eee156c887eed6e25f1ddd0f9add27cd3213531/dsew_community_profile/delphi_dsew_community_profile/pull.py#L750).

This isn't strictly necessary for test positivity, but it makes it easier to match up volume and positivity signals for each publish date.

Empirically, running the pipeline over all reports with a new check for # of timestamps, I don't see any cases where the number of timestamps differs.